File: array-string/remove-duplicates-from-sorted-array-2/main.ts
function removeDuplicates(nums: number[]): number {
  let k = 0;
  let dup = 1;
  for (let i = 1; i < nums.length; i++) {
    if (nums[i] !== nums[k]) {
      k++;
      nums[k] = nums[i];
      dup = 1;
    } else if (dup < 2) {
      k++;
      nums[k] = nums[i];
      dup++;
    }
  }
  k++;
  return k;
}

File: array-string/remove-duplicates-from-sorted-array-2/main.cpp
#include <vector>
using namespace std;

class Solution {
public:
  int removeDuplicates(vector<int> &nums) {
    int n = nums.size();
    int k = 0;
    int duplicates = 1;
    for (int i = 1; i < n; i++) {
      if (nums[i] != nums[k]) {
        k++;
        nums[k] = nums[i];
        duplicates = 1;
      } else if (duplicates < 2) {
        k++;
        nums[k] = nums[i];
        duplicates++;
      }
    }
    k++;
    return k;
  }
};
